test/functional/lua/fs_spec.lua

```diff
@@ -130,6 +131,17 @@ describe('vim.fs', function()
   end)
 
   describe('dir()', function()
+    before_each(function()
+      lfs.mkdir('testd')
+      lfs.mkdir('testd/a')
+      lfs.mkdir('testd/a/b')
+      lfs.mkdir('testd/a/b/c')
```

Comment:
Surprised that using `mkdir_p('testd/a/b/c')` here cause opening the files to fail below